python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,155 @@
+"""
+Table data structure that prints a nicely formatted table
+for an accumulator loop
+"""
+from __future__ import annotations
+
+import inspect
+import sys
+
+import astroid
+import tabulate
+
+
+def num_whitespaces(start_of_loop: str) -> int:
+    """Number of spaces at the beginning of the accumulation loop"""
+    blank_chars = 0
+    for char in start_of_loop:
+        if char.isspace():
+            blank_chars += 1
+        else:
+            break
+
+    return blank_chars
+
+
+def get_loop_lines(lines: list[str], num_whitespace: int) -> str:
+    """Function that returns the lines from the start to the end
+    of the accumulator loop
+    """
+    endpoint = len(lines)
+    for i in range(len(lines)):
+        if lines[i].strip() != "" and not lines[i][num_whitespace].isspace():
+            endpoint = i
+            break
+
+    return "\n".join(lines[:endpoint])
+
+
+def get_for_node(frame: types.FrameType) -> types.NodeNG:
+    """Return the For node from the frame containing the accumulator loop"""
+    func_string = inspect.cleandoc(inspect.getsource(frame))
+    with_stmt_index = inspect.getlineno(frame) - frame.f_code.co_firstlineno
+    lst_str_lines = func_string.splitlines()
+    lst_from_with_stmt = lst_str_lines[with_stmt_index + 1 :]
+    num_whitespace = num_whitespaces(lst_str_lines[with_stmt_index])
+    loop_lines = get_loop_lines(lst_from_with_stmt, num_whitespace)
+
+    return astroid.parse(loop_lines).body[0]
+
+
+class AccumulationTable:
+    """
+    Class used as a form of print debugging to analyze different loop and
+    accumulation variables during each iteration in a for loop
+
+    Private instance attributes:
```

Comment:
You can just call this `Instance attributes`, since not all of them are private.